src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSpark.java

```diff
@@ -0,0 +1,522 @@
+        private static final long serialVersionUID = 1L;
+
+        public final ContigsCollection assembledContigs;
+        public final List<Tuple2<String, Tuple3<ExternalCommandlineProgramModule.ReturnStatus, String, String>>> runtimeInformation;
+
+        public SGAAssemblyResult(final File contigFile, final List<Tuple2<String, Tuple3<ExternalCommandlineProgramModule.ReturnStatus, String, String>>> collectedRuntimeInfo){
+            ContigsCollection proxy = null;
+            try{
+                proxy = new ContigsCollection(contigFile);
+            } catch (final IOException e) { // on IO error caused by converting contigs FASTA file to memory, set in-memory struct to null and append error message
+                collectedRuntimeInfo.add(new Tuple2<>(ContigsCollection.class.getName(),
+                                                      new Tuple3<>(ExternalCommandlineProgramModule.ReturnStatus.STDIOFAIL, "", e.getMessage())));
+            }
+            assembledContigs = proxy;
+            runtimeInformation = collectedRuntimeInfo;
+        }
+    }
+
+    /**
+     * Performs assembly on the FASTA files pointed to by the URI that is associated with the breakpoint identified by the long ID.
+     * Actual assembly work is delegated to other functions.
+     * @param fastqOfABreakpoint    the breakpoint ID and URI to the FASTQ file
+     * @param sgaPath               full path to SGA
+     * @param threads               number of threads to use in various modules, if the module supports parallelism
+     * @param runCorrections        user's decision to run SGA's corrections (with default parameter values) or not
+     * @return                      contig file (if process succeed) and runtime information, associated with the breakpoint ID
+     * @throws IOException          when creating a temporary directory on the local disks for copying the FASTQ file to ,
+     *                              or when copying the FASTQ file from HDFS to local disk.
+     */
+    @VisibleForTesting
+    static Tuple2<Long, SGAAssemblyResult> performAssembly(final Tuple2<Long, URI> fastqOfABreakpoint,
+                                                           final Path sgaPath,
+                                                           final int threads,
+                                                           final boolean runCorrections)
+            throws IOException{
+
+        final Long breakpointId = fastqOfABreakpoint._1();
+        final String pathStringToRawFASTQ = fastqOfABreakpoint._2().getPath();
+        final String rawFASTQFileName = FilenameUtils.getName(pathStringToRawFASTQ);
+
+        final File workingDir = Files.createTempDirectory( "assembly" + breakpointId.toString() ).toAbsolutePath().toFile();
+        workingDir.deleteOnExit();
+
+        BucketUtils.copyFile(pathStringToRawFASTQ, null, workingDir.toPath().toAbsolutePath().toString()+"/"+rawFASTQFileName);
```

Comment:
BucketUtils is a class in the gcs package so I'd expect it to only be necessary when running on google and copying to/from "gs:" paths. I think you will want to use something like `FileSystem.get(ctx.hadoopConfiguration()).copyToLocalFile();`